### PR TITLE
* Removing non-official DI samples

### PR DESCRIPTION
**Pull request by pranavkm**
* Switching to Autofac's nightly feed for unit tests.

Fixes https://github.com/aspnet/DependencyInjection/issues/264

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @Eilon

**Comment by Eilon** [comment]
:shipit:

File: test/Microsoft.Framework.DependencyInjection.Tests/project.json [hunk 1]
```diff
@@ -1,19 +1,17 @@
 {
     "version": "0.1-alpha-*",
     "dependencies": {
+        "Autofac.Framework.DependencyInjection": "4.0.0-*",
         "Microsoft.AspNet.Testing": "1.0.0-*",
         "Microsoft.Framework.DependencyInjection": "1.0.0-*",
-        "Microsoft.Framework.NotNullAttribute.Sources": { "version": "1.0.0-*", "type": "build" },
+        "Microsoft.Framework.NotNullAttribute.Sources": {
```
Comment (davidfowl) [inline test/Microsoft.Framework.DependencyInjection.Tests/project.json]: No bad formatting :smile:

**Comment by davidfowl** [comment]
Do we still need to run unit tests for autofac?

File: test/Microsoft.Framework.DependencyInjection.Tests/project.json (hunk 1 above)
Comment (pranavkm) [inline test/Microsoft.Framework.DependencyInjection.Tests/project.json]: VS 2015 RTM defaults to this. It's a pain to have it switch.

Comment (rynowak) [inline test/Microsoft.Framework.DependencyInjection.Tests/project.json]: We should not fight vs on formatting, it's more work for everyone.

**Comment by Eilon** [comment]
@davidfowl we want to keep them at least for now to make sure we don't break anything.

**Comment by pranavkm** [comment]
@davidfowl the plan was to run tests against the Autofac nightlies so they act as a smoke test. Mvc has tests that run against the Autofac on nuget.org so this should give us reasonable coverage.

**Comment by pranavkm** [comment]
Waiting on this one until https://github.com/aspnet/Mvc/pull/2869 goes in.

**Comment by alexmg** [comment]
I temporarily pulled the unit tests from this project into Autofac for a sanity check against what is essentially the specification. It would be nice to have a dependency that I could pull in that contains these tests, allowing only the `AutofacContainerTests` that returns the populated container as `IServiceProvider` to be present in our project. Something like a `Microsoft.Framework.DependencyInjection.Specification` or `Microsoft.Framework.DependencyInjection.Tests` package for DI container authors.

**Comment by davidfowl** [comment]
@alexmg Great idea. We should make this a package.

**Comment by Eilon** [comment]
Yeah that would be cool. Logged https://github.com/aspnet/DependencyInjection/issues/270 to track this.

**Pull request closed without merging**